core/src/main/java/org/fao/geonet/kernel/metadata/DefaultStatusActions.java

```diff
@@ -221,7 +221,7 @@ public Map<Integer, StatusChangeType> onStatusChange(List<MetadataStatus> listOf
     private boolean applyStatusChange(int metadataId, MetadataStatus status, String toStatusId) throws Exception {
         boolean deleted = false;
         if (!deleted) {
-            metadataStatusManager.setStatusExt(status);
+            metadataStatusManager.setStatusExt(status, false);
```

Comment:
`applyStatusChange` is used in `StatusActions.onStatusChage` that is call when creating a draft version:

https://github.com/geonetwork/core-geonetwork/blob/1f16287e654bc24b6b19b6f3bcd98ef593229676/core/src/main/java/org/fao/geonet/kernel/datamanager/draft/DraftMetadataUtils.java#L592

Previously was indexed the metadata and with this change no. Have you check that creating a draft contains the proper status in the index? Otherwise `DraftMetadataUtils.createDraft` needs to explicit index the draft metadata.

Reply:
I tested the code. This create draft only happens when creating working copy. Its working fine without indexing.

![image](https://github.com/geonetwork/core-geonetwork/assets/74916635/6bd0a0bc-eddb-4103-b6da-8bb3fe33324a)

Reply:
@josegar74 

Sorry I misunderstood your comment previously. You are right, it used to explicit the indexing. So it's better to introduce at its parent level onStatusChange(boolean updateIndex). I made a new commit and please check it out. The only place does not need indexing is 
![image](https://github.com/geonetwork/core-geonetwork/assets/74916635/06dad563-901c-485f-9dec-70b5753e66bf)